File: 1436-destination-city/1436-destination-city.java
class Solution {
    public String destCity(List<List<String>> paths) {
     
        Set<String>sources = new HashSet<>();
        
        for(List<String> path : paths){
            
            String u = path.get(0);
            sources.add(u);
        }
        
        for(List<String> path : paths){
            
            String v = path.get(1);
            if(!sources.contains(v))   return v;
        }
        
        return new String();
        
    }
}

File: 1436-destination-city/1436-destination-city.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        
        unordered_set<string>sources;
        
        for(vector<string> path : paths){
            
            string u = path[0];            
            sources.insert(u);
            
        }
        
        for(vector<string>path : paths){
            
            string v = path[1];            
            if(sources.find(v) == sources.end())    return v;
        }
        
        return "";
        
    }
};